src/python/pants/pantsd/pants_daemon.py

```diff
@@ -493,6 +494,18 @@ def terminate(self, include_watchman=True):
     if include_watchman:
       self.watchman_launcher.terminate()
 
+  def _has_exceeded_memory_usage(self):
+    used_memory = self.current_memory_usage()
```

Comment:
`current_memory_usage` has an assert in it, but it's not obvious that this method is unsafe to call if the process isn't alive. Should this method just return `False` if the process isn't alive...? If not, should document that it "must not be called while the process might be dead" or something.